packages/console/src/pages/ApplicationDetails/ApplicationDetailsContent/ProtectedAppSettings/index.tsx

```diff
@@ -53,10 +53,12 @@ function ProtectedAppSettings({ data }: Props) {
     name: 'protectedAppMetadata.pageRules',
   });
 
-  const host = data.protectedAppMetadata?.host;
-
+  const host = data.protectedAppMetadata?.host ?? '';
   // We only support one custom domain for protected apps at the moment.
   const customDomain = customDomains[0];
+  const externalLink = `https://${
+    customDomain?.status === DomainStatus.Active ? customDomain.domain : host
+  }`;
```

Comment:
empty host should not be allowed?

Reply:
We have the empty check underneath. If the host is empty, then the input will not even be displayed. Technically, if the app is protected app, the host will never be empty, since it requires the host to be filled in when creating the app

Reply:
<img width="423" alt="image" src="https://github.com/logto-io/logto/assets/12833674/f03059d0-a98e-469b-a041-0dc936aba1b9">